python/lsst/cp/pipe/utils.py

```diff
@@ -1032,33 +1131,50 @@ def compute_ratio_model(nodes, group_indices, pars, pd, mu, return_spline=False)
             Array of node positions.
         group_indices : `list` [`np.ndarray`]
             List of group indices, one array for each group.
+        par_indices : `dict`
+            Dictionary showing which indices in the pars belong to
+            each set of fit values.
         pars : `np.ndarray`
             Parameter array, with spline values (one for each node) followed
-            by proportionality constants (one for each group.)
+            by proportionality constants (one for each group); one extra
+            for the offset O (if fit_offset was set to True); two extra
+            for the weights (if fit_weights was set to True); and one
+            extra for the temperature coefficient (if fit_temperature was
+            set to True).
         pd : `np.ndarray` (N,)
             Array of photodiode measurements.
         mu : `np.ndarray` (N,)
             Array of flat means.
+        temperature_scaled : `np.ndarray` (N,)
+            Array of scaled temperature values.
         return_spline : `bool`, optional
             Return the spline interpolation as well as the model ratios?
 
         Returns
         -------
         ratio_models : `np.ndarray` (N,)
-            Model ratio, (mu_i - S(mu_i))/(k_j * D_i)
+            Model ratio, (mu_i - S(mu_i) - O)/(k_j * D_i)
         spl : `lsst.afw.math.thing`
             Spline interpolator (returned if return_spline=True).
         """
         spl = lsst.afw.math.makeInterpolate(
             nodes,
-            pars[0: len(nodes)],
+            pars[par_indices["values"]],
             lsst.afw.math.stringToInterpStyle("AKIMA_SPLINE"),
         )
 
-        numerator = mu - spl.interpolate(mu)
+        # Check if we want to do just the left or both with temp scale.
+        if len(par_indices["temperature_coeff"]) == 1:
+            mu_corr = mu*(1. + pars[par_indices["temperature_coeff"]]*temperature_scaled)
+        else:
+            mu_corr = mu
+
+        numerator = mu_corr - spl.interpolate(mu_corr)
```

Comment:
Does this line contradict the docstrings that list `S(mu_i) + mu_i - O`?  It's also entirely possible I've confused myself.

Reply:
I think I've convinced myself that I am not confused in this case.